text/0002-sarif-result-schema.md

```diff
@@ -0,0 +1,1939 @@
+        name: string;
+        version: string;
+        repository: {
+          totalCommits: number;
+          totalFiles: number;
+          age: string;
+          activeDays: string;
+        };
+      };
+      cli: {
+        schema: number;
+        configHash: string;
+        config: CheckupConfig;
+        version: string;
+        args: {
+          paths: string[];
+        };
+        flags: {
+          config?: string;
+          task?: string[];
+          format: string;
+          outputFile?: string;
+          excludePaths?: string[];
+        };
+        timings: Record<string, number>;
+      };
+      analyzedFiles: string[];
+      analyzedFilesCount: number;
+    };
+    results: TaskResult[];
+    errors: TaskError[];
+    actions: Action[];
+  }
+  ```
+</details>
+
+With SARIF, the results would look like this (there are a ton of optional properties here, so I am keeping only the ones we will need): 
+*Note* I left in the comments from SARIF because I think they are useful to understand the shape of the data here, so when I wanted to add 
+checkup related commentary I added a newline after the existing SARIF comments and prefaced with `CHECKUP:` before adding checkup commentary to differentiate.
+
+<details>
+  <summary>Relevant SARIF data fields</summary>
+
+  ```
+  export interface Log {
+      /**
+      * The URI of the JSON schema corresponding to the version.
+      */
+      $schema?: string;
+
+      /**
+      * The SARIF format version of this log file.
+      */
+      version: Log.version;
+
+      /**
+      * The set of runs contained in this log file. 
+      * 
+      * CHECKUP: this will generally be just one Run
+      */
+      runs: Run[];
+
+      /**
+      * Key/value pairs that provide additional information about the physical location.
+      *
+      * CHECKUP: we can store the project info, analyzedFiles, actions, and analyzedFilesCount here
+      *
+      */
+      properties?: PropertyBag;
+
+  }
+
+  export interface Run {
+      /**
+      * Describes the invocation of the analysis tool. 
+      *
+      * CHECKUP: we will put cli info here
+      */
+      invocations?: Invocation[];
+
+      
+      /**
+      * The set of results contained in an SARIF log. The results array can be omitted when a run is solely exporting
+      * rules metadata. It must be present (but may be empty) if a log file represents an actual scan.
+      */
+      results?: Result[];
+
+      /**
+      * Information about the tool or tool pipeline that generated the results in this run. A run can only contain
+      * results produced by a single tool or tool pipeline. A run can aggregate results from multiple log files, as long
+      * as context around the tool run (tool command-line arguments and the like) is identical for all aggregated files.
+      *
+      * CHECKUP: We may not _need_ to populate this, but I don't see why not :) There is a use case for products using this schema for other 
```

Comment:
We should definitely populate this. This is to be consistent with the SARIF spec, but also to clearly disambiguate the source of the data (from which tool it came).

Reply:
Cool, sg - I am populating it in my implementation, but I will modify this comment